zarr/meta.py

```diff
@@ -187,6 +187,12 @@ def encode_dtype(cls, d: np.dtype):
     @classmethod
     def _decode_dtype_descr(cls, d) -> List[Any]:
         # need to convert list of lists to list of tuples
+
+        # remove metadata in descr
+        if isinstance(d, (list, tuple)) and len(d) == 2 and isinstance(
+                d[1], dict):
+            return d[0]
```

Comment:
Is it possible that this needs the recursive handling of the next section, @ninousf?